File: backend/app/models/destination_model.py
from typing import Any
from uuid import uuid4
from sqlalchemy import BIGINT, Column, Integer, String, Boolean, Float
from app.extensions import db


class DestinationModel(db.Model):
    __tablename__ = "destination"
    id = Column(Integer, primary_key=True, autoincrement=True)
    uid = Column(
        String(64),
    )
    imgUrl = Column(String(128))
    name = Column(String(64))
    city = Column(String(64))
    price = Column(BIGINT)
    rating = Column(Float)
    is_new = Column(Boolean)
    is_popular = Column(Boolean)

    def __init__(
        self,
        *,
        imgUrl: str,
        name: str,
        city: str,
        price: int,
        rating: float,
        isNew: bool,
        isPopular: bool
    ) -> Any:
        self.uid = str(uuid4())
        self.imgUrl = imgUrl
        self.name = name
        self.city = city
        self.price = price
        self.rating = rating
        self.is_new = isNew
        self.is_popular = isPopular

    def save(self):
        db.session.add(self)
        db.session.commit()
        db.session.refresh(self)

File: mobile/lib/models/destination_model.dart
import 'package:equatable/equatable.dart';

class DestinationModel extends Equatable {
  final int? id;
  final String imgUrl;
  final String name;
  final String city;
  final double rating;
  final int? price;
  final bool? isNew;
  final bool? isPopular;

  const DestinationModel({
    required this.id,
    required this.imgUrl,
    required this.name,
    required this.city,
    required this.rating,
    this.price,
    this.isNew,
    this.isPopular,
  });

  factory DestinationModel.fromjson(Map<String, dynamic> json) {
    return DestinationModel(
      id: json['id'],
      imgUrl: json['imgUrl'],
      name: json['name'],
      city: json['city'],
      rating: json['rating'],
      price: json['price'],
      isNew: json['isNew'],
      isPopular: json['isPopular'],
    );
  }
  @override
  List<Object?> get props =>
      [id, imgUrl, name, city, rating, price, isNew, isPopular];
}
